### ISSUE TITLE
"warning: validateDOMNesting(...): <tr> cannot appear as a child of <table>" when using subgrids

### ISSUE THREAD

**Issue by brownieboy** [GriddleGriddle/Griddle#249]
The full error is:

> Warning: validateDOMNesting(...): <tr> cannot appear as a child of <table>. See GridTable > table > GridRowContainer > GridRow > tr. Add a <tbody> to your code to match the DOM tree generated by the browser.

I get this when using subgrids in my table.  Specifically, I have mix of rows in my table, some of which have subgrids and some of which don't.  The rows that have subgrids generate <tr> tags inside <tbody> tags in the resulting HTML.  However, the rows that _don't_ have subgrids generate <tr> tags only (i.e. not inside a <tbody>).  This means that you have <tr> tags that are peers of <tbody> tags. which React doesn't like.

I can test up a test repo for this issue if anybody wants.

I'm using react 0.14 with griddle-react 0.2.14.

**Comment by SylarRuby** [comment]
@brownieboy have you already solved it?

**Comment by brownieboy** [comment]
@SylarRuby,

No, I just lived with it.  It didn't actually cause any problems on the project where I was using it.  It's a warning not an error.

Have you checked in the very latest version?  (I haven't.)

**Comment by lukasjsk** [comment]
@brownieboy @SylarRuby 
We hit the same issue and we were able to get rid of the warning by wrapping rows without children inside <tbody> in gridRowContainter.jsx (last line of render method).

So instead:
```
return that.props.hasChildren === false ? arr[0] : <tbody>{that.state.showChildren ? arr.concat(children) : arr}</tbody>
```
we do
```
return that.props.hasChildren === false ? <tbody>arr[0]</tbody> : <tbody>{that.state.showChildren ? arr.concat(children) : arr}</tbody>
```

This was just a quick fix and I still need to check if it's not breaking anything else...

**Comment by roessera** [comment]
wrapping <tbody>arr[0]</tbody> does not fix the issue when using a grid containing both rows and rows with subgrids (children). I receive the following error upon load: 

Warning: validateDOMNesting(...): <tbody> cannot appear as a child of <tbody>. See GridTable > tbody > GridRowContainer > tbody.

Using React 0.14.5

**Comment by bellatrix1001** [comment]
Has anyone been able to come up with a solution for this?

**Comment by apalanki** [comment]
To tag along with what @lukasjsk proposed, adding 

`return that.props.hasChildren === false ? React.createElement('tbody', null, arr[0]) : React.createElement('tbody', null, that.state.showChildren ? arr.concat(children) : arr);`

fixed this issue for me without breaking anything on React 15.3.1.
